### ISSUE TITLE
Bug: When ingesting a File entity its @id gets a # suffix

### ISSUE THREAD

**Issue by ptsefton** [ResearchObject/ro-crate-py#77]
If I ingest the below crate and then inspect the @id then  {"@id": "test.csv", "@type": "File"} turns into 

{'@id': '#test.csv', '@type': 'File'}

I have tried this with the file in the directory and without - same result.

Using code like this:

```
crate = ROCrate("./") 
for e in crate.get_entities():
    print(e.as_jsonld())  # JSON entry

```

```
{
  "@context": [
    "https://w3id.org/ro/crate/1.1/context",
    {
      "@vocab": "http://schema.org/"
    },
    {
      "@base": null
    }
  ],
  "@graph": [
    {
      "@id": "#collection",
      "@type": "RepositoryCollection ",
      "name": "Test collection"
    },
    {
      "@id": "./",
      "@type": "Dataset",
      "hasFile": [{"@id": "test.csv"}],
      "hasPart": [
        {
          "@id": "#collection"
        }
      ],
      "name": "testing hasPart"
    },
    {"@id": "test.csv", "@type": "File"},
    {
      "@id": "ro-crate-metadata.json",
      "@type": "CreativeWork",
      "about": {
        "@id": "./"
      },
      "identifier": "ro-crate-metadata.json"
    }


  ]
}
```

**Comment by simleo** [comment]
The spec says [data entities MUST be linked from the root via `hasPart`](https://www.researchobject.org/ro-crate/1.1/data-entities.html#referencing-files-and-folders-from-the-root-data-entity); `hasFile`, OTOH, is only mentioned as a term imported from PCDM. When reading a crate, the library adds all items listed in `hasPart` as data entities, then assumes everything else is a contextual entity. Since `"test.csv"` is not an absolute URI, it adds a `#` to turn it into a local id.

To sum it up, I believe the library is behaving consistently with the spec here. I will close this and perhaps you can open an issue on the ro-crate repo if you think a spec update is necessary.
